aws-datastore/src/test/java/com/amplifyframework/datastore/storage/InMemoryStorageAdapter.java

```diff
@@ -160,23 +160,30 @@ public void initialize(
             @NonNull final Consumer<DataStoreException> onError
     ) {
         final int index = indexOf(item);
-        if (index > -1 && (predicate == null || predicate.evaluate(item))) {
-            Model savedItem = items.remove(index);
-            StorageItemChange.Record deletion = StorageItemChange.<T>builder()
-                    .item((T) savedItem)
-                    .itemClass((Class<T>) savedItem.getClass())
-                    .type(StorageItemChange.Type.DELETE)
-                    .predicate(predicate)
-                    .initiator(initiator)
-                    .build()
-                    .toRecord(storageItemChangeConverter);
-            changeRecordStream.onNext(deletion);
-            onSuccess.accept(deletion);
-        } else {
+        if (index < 0) {
             onError.accept(new DataStoreException(
-                    "Item does not exist or conditional check failed.",
-                    "Verify the saved models."));
+                    "This item was not found in the datastore: " + item.toString(),
+                    "Use save() function to create models to store."
```

Comment:
I think you need a `return;` here, too, afteryou fire `onError.accept(...)`.

Reply:
AHA! pushed the change right before this comment ;)